WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductSelectionListItemLookup.kt

```diff
@@ -0,0 +1,23 @@
+package com.woocommerce.android.ui.products
+
+import android.view.MotionEvent
+import androidx.recyclerview.selection.ItemDetailsLookup
+import androidx.recyclerview.widget.RecyclerView
+
+/**
+ * Class that provides the recyclerview selection library the information about the
+ * items associated with the users selection.
+ *
+ * That selection is based on a MotionEvent that we have mapped to the [ProductItemViewHolder].
+ * Given that motion event, we will find the child of the RecyclerView where the event happened and
+ * return the details of that item.
+ */
+class ProductSelectionListItemLookup(private val recyclerView: RecyclerView) : ItemDetailsLookup<Long>() {
+    override fun getItemDetails(event: MotionEvent): ItemDetails<Long>? {
+        val view = recyclerView.findChildViewUnder(event.x, event.y)
+        if (view != null) {
+            return (recyclerView.getChildViewHolder(view) as? ProductItemViewHolder)?.getItemDetails()
+        }
+        return null
+    }
```

Comment:
np: Since this method interface allows nullable returns and this cast operation can also return `null`, shouldn't we just return directly the result of the transformation instead of check if the `view` is null and then return a literal `null` value or a nullable operation?
```suggestion
    override fun getItemDetails(event: MotionEvent): ItemDetails<Long>? =
        recyclerView
            .findChildViewUnder(event.x, event.y)
            ?.let { recyclerView.getChildViewHolder(it) as? ProductItemViewHolder }
            ?.getItemDetails()
```